preppal-be/routes/reviewApi.ts

```diff
@@ -0,0 +1,129 @@
+const expressReviewApi = require("express");
+const configReviewApi = require("../configs/secrets.ts");
+
+const Review = require("../models/review.ts");
+const Author = require("../models/user.ts");
+const Recipe = require("../models/recipe.ts");
+
+const routerReviewApi = expressReviewApi.Router();
+
+
+/**
+ * GET - Get all recipes
+ */
+routerReviewApi.get("/", async (req, res) => {
+    try {
+        const reviews = await Review.find();
+        res.status(200).json(reviews);
+    }
+    catch (error) {
+        console.error(error.message);
+        res.status(500).send("Server error.");
+    }
+});
+
+/**
+ * GET - Get reviews by recipeId
+ */
+routerReviewApi.get("/:recipeId", async (req, res) => {
+    try {
+        const reviews = await Review.findOne({ recipeId: req.params.recipeId });
+        res.status(200).json(reviews);
+    }
+    catch (error) {
+        console.error(error.message);
+        res.status(500).send("Server error.");
+    }
+});
+
+/**
+ * GET - Init the review for a recipe
+ */
+routerReviewApi.get("/new/:recipeId", async (req, res) => {
+    try {
+        const recipeId = req.params.recipeId;
+
+        const reviews = await Review.findOne({ recipeId });
+        const recipe = await Recipe.findOne({ _id: recipeId });
+
+        if (reviews) {
```

Comment:
does this work? Doesn't this find if reviews exist and if so cannot add more reviews?

Reply:
unless this is just creating an entry for recipe reviews in our database, but I don't know if it should send a 400 error. Up to you

Reply:
We create a new table in the database for reviews. Each recipe is pretty much only holds an array of reviews.
This endpoint is only called at the same time when we create a new recipe. It was also used to manually create these new tables for older recipes.

So how this works is that it first checks if this table exists for a review. Then it checks to make sure the recipe exists. If those a review doesn't exist and the recipe does, it creates a new table. We do this just so we don't duplicate review sections for each recipe.